services/search/search_engine.py

```diff
@@ -41,6 +41,7 @@ def query(self, query):
         logging.debug("keywords location: %s", env.get_keyword_endpoint())
 
         response = requests.post(env.get_keyword_endpoint(), json=query_text)
+        response.raise_for_status()
         response = response.json()
```

Comment:
[nit] technically unrelated to this PR, but overwriting a variable with a different object type (`.json()` returns a string) is a weird python thing, and probably bad practice as it can lead to confusion to people reading the code who only noticed one of the assignments to `response`

Reply:
I just assumed my reviewers are perfect beings